bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/part/MultiPageEditorPart.java

```diff
@@ -334,13 +334,13 @@ public void createPartControl(Composite parent) {
 		// done
 		if (getActivePage() == -1) {
 			setActivePage(0);
-			IEditorPart part = getEditor(0);
-			if (part != null) {
-				final IServiceLocator serviceLocator = part.getEditorSite();
-				if (serviceLocator instanceof INestable) {
-					activeServiceLocator = (INestable) serviceLocator;
-					activeServiceLocator.activate();
-				}
+		}
+		IEditorPart part = getEditor(0);
```

Comment:
This doesn't look right now. Comment above says "unless it has already been done". So this code should now use `getEditor(getActivePage())`, otherwise the wrong service locator may get activated.

Can that service locator of the active page be already activated here?

Reply:
> This doesn't look right now. Comment above says "unless it has already been done". So this code should now use `getEditor(getActivePage())`, otherwise the wrong service locator may get activated.


I totally agree, it should be this way to make it more appropriate. Get the present active page and proceed.
`IEditorPart part = getEditor(getActivePage());`

Thanks for your review.